google-cloud-logging/src/main/java/com/google/cloud/logging/MonitoredResourceUtil.java

```diff
@@ -160,7 +164,11 @@ private static String getValue(Label label, String resourceType) {
       case ContainerName:
         if (resourceType.equals("k8s_container")) {
           String hostName = System.getenv("HOSTNAME");
-          value = hostName.substring(0, hostName.indexOf("-"));
+          if (hostName == null) {
+            value = "unknown";
```

Comment:
This logic should be reverted on its head: first asking metadata endpoint, then if exception, null or empty => ask env variable (HOSTNAME), then trying to figure out the container name from host name. Also probably for better readability would be better to just pass the value of hostName and make the determination of the value inside of getContainerNameFromHostName. This way there are no two places for the logic.

Reply:
I'm asking metadata but afaics it just returns null if there's any exception, so I'm checking that instead of catching exception. Also, I'm not just blanket getting the hostname, but only when it's a k8s_container.
Alternatively I could just return the hostname if not k8s_container and do the logic in K8sContainerUtil if it is a k8s. wdyt?